### ISSUE TITLE
Add scripts to `lint-es` script

### ISSUE THREAD

**Issue by BSFishy** [opensearch-project/oui#1096]
Currently, `lint-es` only lints `src` and `src-docs`. It would be nice to also have the `scripts` directory linted by this script too.

**Comment by MadaniKK** [comment]
Hi, can I take this issue? Thank you!

**Comment by BSFishy** [comment]
> Hi, can I take this issue? Thank you!

Sure! This should be pretty easy. Feel free to ping if you run into any issues!

**Comment by MadaniKK** [comment]
I

> > Hi, can I take this issue? Thank you!
> 
> Sure! This should be pretty easy. Feel free to ping if you run into any issues!

Is it just adding one line in the .sass_lint.yml? I am not very familiar with linter..

**Comment by BSFishy** [comment]
Nope, this should just be updating the `lint-es` script in `package.json`:

https://github.com/opensearch-project/oui/blob/ae0171d5fd0ffb9237367d2d8e351be9d4638623/package.json#L25

Currently we are checking directories `{src,src-docs}`, but we want `{src,src-docs,scripts}`. This will also most likely involve running `yarn lint-es-fix`, as we previously weren't linting that folder, so there is most likely some places where the linter will snag

**Comment by MadaniKK** [comment]
Ah, that makes sense! Thank you for your instructions!
Also right now running `yarn lint-es --fix` returned some errors. It looks like the .js files in `scripts/` should be using import statements instead of `require`. Should I disable this specific es-lint rule or should I change all the `require`s?
<img width="735" alt="image" src="https://github.com/opensearch-project/oui/assets/71857976/28fb6f80-3063-4c81-858a-1e242c5a703f">

**Comment by BSFishy** [comment]
> Should I disable this specific es-lint rule or should I change all the `require`s?

I don't think we want to disable the ESLint rule, as that will disable it for the entire project, which we don't want. Why don't you test and see if the import statement will work? I'm usually a little scared to use import statements in scripts run directly by Node, but we should be at a late enough version that they work as expected.

**Comment by MadaniKK** [comment]
> I don't think we want to disable the ESLint rule, as that will disable it for the entire project, which we don't want. Why don't you test and see if the import statement will work? I'm usually a little scared to use import statements in scripts run directly by Node, but we should be at a late enough version that they work as expected.

I tried changing them into import, but it looks like it will break a lot of things when running `yarn start`..... I am using node 18.18.0... I tried setting the `package.json` file   "type": "module" to use ESM... but that will break other things too
So now I am not sure what to do. 
<img width="880" alt="image" src="https://github.com/opensearch-project/oui/assets/71857976/04754a4c-c30b-4ee8-b387-6a8949741304">

**Comment by BSFishy** [comment]
Yeah I was worrying we would see something like that. Not sure about how to go forward in that case... Looping in @AMoo-Miki and @joshuarrrr

**Comment by BSFishy** [comment]
Summary from office hours: let's convert the imports back into the `require` syntax and figure out how to conditionally suppress the `@typescript-eslint/no-var-requires` for the scripts directory.

**Comment by anirudha** [comment]
Closing — this issue has been open for 1-2+ years without progress. If still relevant, please reopen or create a new issue with updated context.
